nav2_motion_primitives/src/back_up.cpp

```diff
@@ -79,15 +75,15 @@ nav2_tasks::TaskStatus BackUp::controlledBackup()
 
   double diff_x = initial_pose_->pose.pose.position.x - current_odom_pose->pose.pose.position.x;
   double diff_y = initial_pose_->pose.pose.position.y - current_odom_pose->pose.pose.position.y;
-  double distance = sqrt(diff_x * diff_x + diff_y * diff_y);
+  double distance_2 = diff_x * diff_x + diff_y * diff_y;
 
-  if (distance >= abs(command_x_)) {
+  if (distance_2 >= abs(command_x_2_)) {
     cmd_vel.linear.x = 0;
     robot_->sendVelocity(cmd_vel);
     return TaskStatus::SUCCEEDED;
   }
   // TODO(mhpanah): cmd_vel value should be passed as a parameter
-  command_x_ < 0 ? cmd_vel.linear.x = -0.025 : cmd_vel.linear.x = 0.025;
+  command_x_2_ < 0 ? cmd_vel.linear.x = -0.025 : cmd_vel.linear.x = 0.025;
```

Comment:
This is always going to be false since command_x_2, being a square, should always be positive ... well, unless we allow imaginary values as an input ;-)

Reply:
this.... is a good point. I will revery back to not trying to get rid of the square root.